test/src/common/poster.spec.js

```diff
@@ -0,0 +1,254 @@
+            done();
+          } catch (e) {
+            done(e);
+          }
+        })
+        .catch(e => {
+          done(e);
+        });
+    });
+
+    it('should choose backend poster', function(done) {
+      kalturaPlayer = setup(config);
+      sandbox.stub(kalturaPlayer._provider, 'getMediaConfig').resolves(mediaConfig1);
+      kalturaPlayer
+        .loadMedia({entryId: entryId})
+        .then(() => {
+          try {
+            kalturaPlayer.config.sources.poster.should.have.string(mediaConfig1.sources.poster);
+            done();
+          } catch (e) {
+            done(e);
+          }
+        })
+        .catch(e => {
+          done(e);
+        });
+    });
+
+    it('should choose backend poster on change media', function(done) {
+      kalturaPlayer = setup(config);
+      sandbox.stub(kalturaPlayer._provider, 'getMediaConfig').resolves(mediaConfig1);
+      kalturaPlayer
+        .loadMedia({entryId: entryId})
+        .then(() => {
+          try {
+            kalturaPlayer.config.sources.poster.should.have.string(mediaConfig1.sources.poster);
+          } catch (e) {
+            done(e);
+            return;
+          }
+          sandbox.restore();
+          sandbox.stub(kalturaPlayer._provider, 'getMediaConfig').resolves(JSON.parse(JSON.stringify(mediaConfig2)));
+          kalturaPlayer
+            .loadMedia({entryId: alterEntryId})
+            .then(() => {
+              try {
+                kalturaPlayer.config.sources.poster.should.have.string(mediaConfig2.sources.poster);
+                done();
+              } catch (e) {
+                done(e);
+              }
+            })
+            .catch(e => {
+              done(e);
+            });
+        })
+        .catch(e => {
+          done(e);
+        });
+    });
+
+    it('should choose configured poster on change media', function(done) {
+      kalturaPlayer = setup(config);
+      sandbox.stub(kalturaPlayer._provider, 'getMediaConfig').resolves(JSON.parse(JSON.stringify(mediaConfig1)));
+      kalturaPlayer
+        .loadMedia({entryId: entryId})
+        .then(() => {
+          try {
+            kalturaPlayer.config.sources.poster.should.have.string(mediaConfig1.sources.poster);
+          } catch (e) {
+            done(e);
+            return;
+          }
+          kalturaPlayer.reset();
+          kalturaPlayer.configure({
+            sources: {
+              poster: myCustomPosterUrl
+            }
+          });
+          kalturaPlayer
+            .loadMedia({entryId: alterEntryId})
+            .then(() => {
+              try {
+                kalturaPlayer.config.sources.poster.should.equal(myCustomPosterUrl);
+                done();
+              } catch (e) {
+                done(e);
+              }
+            })
+            .catch(e => {
+              done(e);
+            });
+        })
+        .catch(e => {
+          done(e);
+        });
+    });
+  });
+});
+
+const mediaConfig1 = {
```

Comment:
Can we use the media mock-data like in the file below?

Reply:
done